### PR TITLE
Migrate to GitHub Actions (melodic-devel)

### PR DESCRIPTION
**Pull request by tylerjw**
what the title says, like the others

### CODE REVIEW

File: .github/workflows/industrial_ci_action.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+name: BuildAndTest
+
+on: [push, pull_request]
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - {ROS_DISTRO: melodic, ROS_REPO: main}
```
Comment (rhaschke) [inline .github/workflows/industrial_ci_action.yml]: `catkin_lint` is [part of the main build job](https://github.com/tylerjw/warehouse_ros_mongo/runs/2150869355?check_suite_focus=true), no need to test this twice. Instead go for Kinetic as before with Travis:
```suggestion
          - {ROS_DISTRO: kinetic, ROS_REPO: main}
```

**Review by rhaschke — APPROVED** [review]
Small nit, otherwise looks good.

File: .github/workflows/industrial_ci_action.yml (hunk 1 above)
Comment (tylerjw) [reply .github/workflows/industrial_ci_action.yml]: Pushed a rebase of this with that change.

Comment (rhaschke) [reply .github/workflows/industrial_ci_action.yml]: Aren't you missing `catkin_lint` now?

**Review by rhaschke — CHANGES_REQUESTED** [review]

**Comment by tylerjw** [comment]
> Aren't you missing catkin_lint now?

I thought you meant to not do catkin_lint here (since it is done on master).  I'll put it back, I misunderstood.  I'm trying to fix the issue with kinetic and the before script first.  I'll rebase and add back catkin_lint once I figure this out.

**Comment by rhaschke** [comment]
Sorry for being ambiguous. You had two build jobs:
* melodic, main, catkin_lint
* melodic, main
Here the second is redundant.

**Comment by tylerjw** [comment]
Ok, I believe this is finally ready.  I'm sorry about the confusion.  I got it working installing mongo's interface from source on kinetic so that run passes too.

File: .github/workflows/industrial_ci_action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - {ROS_DISTRO: melodic, ROS_REPO: main, CATKIN_LINT: pedatntic}
+          - {
+              ROS_DISTRO: kinetic, ROS_REPO: main, 
+              BEFORE_BUILD_TARGET_WORKSPACE: './.ci.before_script', 
+              ROSDEP_SKIP_KEYS: 'libmongoclient-dev',
+              ADDITIONAL_DEBS: 'libssl-dev scons git'
```
Comment (rhaschke) [inline .github/workflows/industrial_ci_action.yml:18]: Nice!

**Review by rhaschke — APPROVED** [review]

**Pull request merged**